t/cmd/lfstest-count-tests.go

```diff
@@ -0,0 +1,270 @@
+)
+
+// countFn is a type signature that all functions who wish to modify the
+// test_count must inhabit.
+//
+// The first and only formal parameter is the current number of running tests
+// found in test_count after acquiring the lock.
+//
+// The returned tuple indicates (1) the new number that should be written to
+// test_count, and (2) if there was an error in computing that value. If err is
+// non-nil, the program will exit and test_count will not be updated.
+type countFn func(int) (int, error)
+
+func main() {
+	if len(os.Args) > 2 {
+		fmt.Fprintf(os.Stderr,
+			"usage: %s [increment|decrement]\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	ctx, cancel := context.WithTimeout(
+		context.Background(), lockAcquireTimeout)
+	defer cancel()
+
+	if err := acquire(ctx); err != nil {
+		fatal(err)
+	}
+	defer release()
+
+	if len(os.Args) == 1 {
+		// Calling with no arguments indicates that we simply want to
+		// read the contents of test_count.
+		callWithCount(func(n int) (int, error) {
+			fmt.Fprintf(os.Stdout, "%d\n", n)
+			return n, nil
+		})
+		return
+	}
+
+	var err error
+
+	switch strings.ToLower(os.Args[1]) {
+	case "increment":
+		err = callWithCount(func(n int) (int, error) {
+			if n > 0 {
+				// If n>1, it is therefore true that a
+				// lfstest-gitserver invocation is already
+				// running.
+				//
+				// Hence, let's do nothing here other than
+				// increase the count.
+				return n + 1, nil
+			}
+
+			// The lfstest-gitserver invocation (see: below) does
+			// not itself create a gitserver.log in the appropriate
+			// directory. Thus, let's create it ourselves instead.
+			log, err := os.Create(fmt.Sprintf(
+				"%s/gitserver.log", os.Getenv("LFSTEST_DIR")))
+			if err != nil {
+				return n, err
+			}
+
+			// The executable name depends on the X environment
+			// variable, which is set in script/cibuild.
+			var cmd *exec.Cmd
+			if runtime.GOOS == "windows" {
+				cmd = exec.Command("lfstest-gitserver.exe")
+			} else {
+				cmd = exec.Command("lfstest-gitserver")
+			}
+
+			// The following are ported from the old
+			// test/testhelpers.sh, and comprise the requisite
+			// environment variables needed to run
+			// lfstest-gitserver.
+			cmd.Env = append(os.Environ(),
+				fmt.Sprintf("LFSTEST_URL=%s", os.Getenv("LFSTEST_URL")),
+				fmt.Sprintf("LFSTEST_SSL_URL=%s", os.Getenv("LFSTEST_SSL_URL")),
+				fmt.Sprintf("LFSTEST_CLIENT_CERT_URL=%s", os.Getenv("LFSTEST_CLIENT_CERT_URL")),
+				fmt.Sprintf("LFSTEST_DIR=%s", os.Getenv("LFSTEST_DIR")),
+				fmt.Sprintf("LFSTEST_CERT=%s", os.Getenv("LFSTEST_CERT")),
+				fmt.Sprintf("LFSTEST_CLIENT_CERT=%s", os.Getenv("LFSTEST_CLIENT_CERT")),
+				fmt.Sprintf("LFSTEST_CLIENT_KEY=%s", os.Getenv("LFSTEST_CLIENT_KEY")),
+			)
+			cmd.Stdout = log
+
+			// Start performs a fork/execve, hence we can abandon
+			// this process once it has started.
+			if err := cmd.Start(); err != nil {
+				return n, err
+			}
+			return n + 1, nil
```

Comment:
Maybe we should return a hard-coded `1` here as below?

Reply:
Sounds like a good idea to me; this makes everything nice and consistent. I got that as: bbb8f5e4 (t/cmd/lfstest-count-tests.go: prefer consistent constant usage, 2018-07-16)/